examples/passport-login/src/authentication-interceptors/oauth2.ts

```diff
@@ -52,7 +52,7 @@ export class CustomOauth2 implements Provider<Interceptor> {
       const request = requestCtx.request;
       // if provider name is given then use this middleware otherwise pass along
       if (request.query['oauth2-provider-name'] === 'oauth2') {
-        return toInterceptor(passport.authenticate('oauth2'))(
+        return toInterceptor(passport.authenticate('facebook'))(
```

Comment:
Please move this function out as we don't want to call `passport.authenticate` multiple times.

Reply:
In the code below, we probably should use `return next();` instead of just `next();`.

Reply:
agreed